### PR TITLE
Make libMLIR_h automatically regenerate

### PR DESCRIPTION
**Pull request by Pangoraw**

### CODE REVIEW

**Comment by Pangoraw** [comment]
The generation is missing stablehlo header files which are needed before merging

**Comment by wsmoses** [comment]
@mofeing honestly I’m confused why the api calls weren’t here to begin with.

when these were in the Yggdrasil build it should’ve exported every symbol we built into Julia headers so that implies to be that the header rn is wrong somehow…

**Comment by Pangoraw** [comment]
so essentially replace this line: 

https://github.com/EnzymeAD/Reactant.jl/blob/a4f685cad9514a0d31b0df38658dafb5ce46b8c9/src/mlir/MLIR.jl#L12

with 

```julia
include("path/to/Reactant_jll/libMLIR_h.jl")
```
?

**Comment by mofeing** [comment]
Wait, you should just rebuild "libMLIR_h.jl" locally by running:

```
bazel build //:libMLIR_h.jl
```

I just forgot to include that one in the `make-dialects.jl` script, so it regenerates automatically.

**Comment by Pangoraw** [comment]
Ok so the new `make-dialects.jl` should be enough ? it automatically runs on each pr ?

File: src/mlir/IR/Type.jl [hunk 1]
```diff
@@ -90,7 +90,7 @@ Type(T::Core.Type{<:Integer}; context::Context=context()) =
 
 Creates a complex type with the given element type.
 """
-Type(::Core.Type{Complex{T}}; context=context()) where {T} = Type(API.mlirComplexTypeGet(Type(T)))
+Type(T::Core.Type{<:Complex}; context=context()) = Type(API.mlirComplexTypeGet(Type(T(im) |> real |> typeof)))
```
Comment (mofeing) [inline src/mlir/IR/Type.jl:93]: This shouldn't be in this PR right?

File: deps/ReactantExtra/make-dialects.jl [hunk 2]
```diff
@@ -16,7 +8,24 @@ for file in [
     )
     Base.Filesystem.cp(
         joinpath(@__DIR__, "bazel-bin", file),
-        joinpath(dirname(dirname(@__DIR__)), "src", "mlir", "Dialects", file);
+        output_path;
         force=true,
     )
 end
+
+src_dir = joinpath(@__DIR__, "..", "..", "src")
```
Comment (mofeing) [inline deps/ReactantExtra/make-dialects.jl:16]: It's better if you use `dirname` instead of `".."` so that it works in non-posix systems like Windows.

**Review by mofeing — CHANGES_REQUESTED** [review]
This should be it. About your question, no: it doesn't run on each PR but on periodically every night and by manual dispatch. If you want, you can change the workflow to run on every PR but on 99% of the PRs it will do nothing.

EDIT: You can add a filter to the workflow such that it automatically runs on PRs that modify ReactantExtra.

File: src/mlir/IR/Type.jl (hunk 1 above)
Comment (Pangoraw) [reply src/mlir/IR/Type.jl]: #199

**Review by mofeing — APPROVED** [review]

**Pull request merged**